Automattic/Gravatar-SDK-Android
# QuickEditor: Verify authorized account email matches with app's email

Closes #306 

### Description

This PR enhances the OAuth flow to ensure the user has authorized the app with the correct email address. It implements a validation step using the new `/me/associated-email` endpoint to verify that the email used for authorization matches the one associated with the user's QE account. This change prevents potential mismatches when the user is already logged in with a different email.

**`Note:`** ~~Currently, we are handling that error like a generic OAuth flow. However, we may want to create a spefic case for it. We need to discuss this with the designers.~~
_Edited_: As discussed we should show an specific error for this case along with the "retry" (`Log in`) button

```
Title: Ooops

Message: It looks like you used the wrong email to log in. Please try again using email@domain.com this time. Thanks!

Button: "Log in"
```

### Testing Steps

**Note:** These testing steps assume you are not already logged in to the browser with any Gravatar account.
** Note 2: ** You'll need two Gravatar accounts. Let's say accounts `A` and `B`.

1. Navigate to the `Avatar Update` tab in the `demo-app`
2. With your account `A` click on the `Update Avatar` button
3. Follow the OAuth process with the account `A`
4. You should see the **Avatars Picker**, tap `Done` button
5. Fill the `Gravatar E-Mail` with the account `B`
6. Click the `Update Avatar` button
7. You should see the OAuth approval screen, but with `A` account
8. Tap on the `Approve` button
9. QE bottom sheet should show the specific error for this case
10. Tap over the `Log in` button
11. You should see the OAuth approval screen, but with `A` account
12. Tap on the `switch account` link
13. Follow the OAuth process with the account `B`
14. Tap on the `Approve` button
15. You should see the **Avatars Picker**
